devito/finite_differences/tools.py

```diff
@@ -261,3 +261,22 @@ def generate_indices_staggered(func, dim, order, side=None, x0=None):
             ind = [start, start - diff]
 
     return start, tuple(ind)
+
+
+def make_get_shifted_oringin(shift, shape):
+    """
+    Returns a callable that calculates a shifted origin for each derivative
+    of a given operation derivatives scheme (shape) and a shift object which
```

Comment:
I don't think what you call shape here is the shape in the devito sense. 

This is what we call `shape` in devito: https://github.com/devitocodes/devito/blob/master/devito/types/tensor.py#L42

Probably what you're after here is the dimension of the space domain, ie `(3,)` if `f(x, y, z)` or `(1, 1)` in `[f(x), g(x)]` (here a VectorFunction). Am I correct? (@mloubout )

Reply:
`(3, 3)` for `[f(x), g(x)]` if they have `(x, y, z)` dimensions. Yes, you're right. "shape" here does not correspond to Devito's nomenclature. Should we call it `ndim`?

Reply:
yeah, I think so